From c7656af901c68006c3a82478c56a4c9cf141c4ea Mon Sep 17 00:00:00 2001
From: Cheyenne Wills <cwills@sinenomine.net>
Date: Sun, 1 Jun 2025 06:57:55 -0600
Subject: [PATCH 03/18] Linux: Add required MODULE_DESCRIPTION

The Linux 6.15 commit:
    "modpost: require a MODULE_DESCRIPTION()" (6c6c1fc09de3)
changed the Linux build process to require the MODULE_DESCRIPTION.

Add a MODULE_DESCRIPTION() to the kernel modules, including the code
template used in autoconf tests.

Reviewed-on: https://gerrit.openafs.org/16372
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Michael Meffie <mmeffie@sinenomine.net>
Reviewed-by: Andrew Deason <adeason@sinenomine.net>
(cherry picked from commit f2705de3ded4e194bd447d88013076e707ff33d1)

Change-Id: I7fefbb6c1960948dee6424f11789af19b07d3e84
Reviewed-on: https://gerrit.openafs.org/16418
Reviewed-by: Mark Vitale <mvitale@sinenomine.net>
Reviewed-by: Andrew Deason <adeason@sinenomine.net>
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Benjamin Kaduk <kaduk@mit.edu>
(cherry picked from commit e76b520f1316e7059a7855078f117ce194734317)
---
 src/afs/LINUX/osi_module.c     | 1 +
 src/afs/LINUX/osi_pag_module.c | 1 +
 src/cf/linux-test1.m4          | 1 +
 3 files changed, 3 insertions(+)

diff --git a/src/afs/LINUX/osi_module.c b/src/afs/LINUX/osi_module.c
index 0fc9beba0..d3b00af70 100644
--- a/src/afs/LINUX/osi_module.c
+++ b/src/afs/LINUX/osi_module.c
@@ -136,6 +136,7 @@ afs_cleanup(void)
 }
 
 MODULE_LICENSE("http://www.openafs.org/dl/license10.html");
+MODULE_DESCRIPTION("OpenAFS AFS filesystem client");
 module_init(afs_init);
 module_exit(afs_cleanup);
 
diff --git a/src/afs/LINUX/osi_pag_module.c b/src/afs/LINUX/osi_pag_module.c
index 7e8e4f15b..6442077d5 100644
--- a/src/afs/LINUX/osi_pag_module.c
+++ b/src/afs/LINUX/osi_pag_module.c
@@ -107,6 +107,7 @@ afspag_cleanup(void)
 }
 
 MODULE_LICENSE("http://www.openafs.org/dl/license10.html");
+MODULE_DESCRIPTION("OpenAFS standalone PAG manager");
 module_init(afspag_init);
 module_exit(afspag_cleanup);
 
diff --git a/src/cf/linux-test1.m4 b/src/cf/linux-test1.m4
index 010a4464f..581dc10e2 100644
--- a/src/cf/linux-test1.m4
+++ b/src/cf/linux-test1.m4
@@ -23,6 +23,7 @@ $2
 } 
 
 MODULE_LICENSE("http://www.openafs.org/dl/license10.html");
+MODULE_DESCRIPTION("openafs conftest");
 _ACEOF
     echo make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir $SPARSE_MAKEFLAGS modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD &&
     make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir $SPARSE_MAKEFLAGS modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD 2>conftest.err &&
-- 
2.51.0

